src/main/java/easytests/api/v1/controllers/SubjectsController.java

```diff
@@ -66,7 +68,26 @@
     /**
      * show(subjectId)
      */
-    /**
-     * delete(subjectId)
-     */
+
+    @DeleteMapping("/{subjectId}")
+    public void delete(@PathVariable Integer subjectId) throws NotFoundException, ForbiddenException {
+        final SubjectModelInterface subjectModel = this.getSubjectModel(subjectId);
+        if (subjectModel == null) {
+            throw new NotFoundException();
+        }
+        if (!this.acl.hasAccess(subjectModel)) {
+            throw new ForbiddenException();
+        }
+        this.subjectsService.delete(new SubjectsOptions().withRelations(subjectModel));
+    }
+
+    private SubjectModelInterface getSubjectModel(Integer id, SubjectsOptionsInterface subjectOpt) {
+        final SubjectModelInterface subjectModel = this.subjectsService.find(id, subjectOpt);
+        return subjectModel;
+    }
+
+    private SubjectModelInterface getSubjectModel(Integer id) throws NotFoundException {
```

Comment:
I don't think that getSubjectModel mathod is good for fething it for delete.
Use getSubject as always - for Auth.

And if you want to fetch subject for delete manually pass this.subjectsOptionsBuilder.forDelete() into ` getSubjectModel(Integer id, SubjectsOptionsInterface subjectOpt)` in delete method.